python/igraphalg.py

```diff
@@ -0,0 +1,290 @@
+
+    return mgp.Record(
+        nodes=convert_vertex_ids_to_mgp_vertices(sorted, ctx, inverted_id_mapping)
+    )
+
+
+@mgp.read_proc
+def community_leiden(
+    ctx: mgp.ProcCtx,
+    edge_property: mgp.Nullable[str] = None,
+    resolution_parameter: float = 0.6,
+    number_of_iterations: int = -1,
+) -> mgp.Record(community_id=int, community_members=List[mgp.Vertex]):
+    graph, _, inverted_id_mapping = create_igraph_from_ctx(ctx, directed=False)
+
+    communities = graph.community_leiden(
+        resolution_parameter=resolution_parameter,
+        weights=edge_property,
+        n_iterations=number_of_iterations,
+    )
+
+    return [
+        mgp.Record(
+            community_id=i,
+            community_members=convert_vertex_ids_to_mgp_vertices(
+                members, ctx, inverted_id_mapping
+            ),
+        )
+        for i, members in enumerate(communities)
+    ]
+
+
+@mgp.read_proc
+def spanning_tree(
+    ctx: mgp.ProcCtx, edge_property: mgp.Nullable[str] = None, directed: bool = True
+) -> mgp.Record(tree=List[List[mgp.Vertex]]):
+    graph, id_mapping, inverted_id_mapping = create_igraph_from_ctx(
+        ctx, directed=directed
+    )
+    edge_weights = None
+    if edge_property:
+        edge_weights = graph.es[edge_property]
+
+    min_span_tree_graph = graph.spanning_tree(weights=edge_weights)
+    min_span_tree = get_min_span_tree_vertex_pairs(
+        ctx=ctx,
+        id_mapping=id_mapping,
+        inverted_id_mapping=inverted_id_mapping,
+        min_span_tree_graph=min_span_tree_graph,
+    )
+
+    return mgp.Record(tree=min_span_tree)
+
+
+@mgp.read_proc
+def shortest_path_length(
+    ctx: mgp.ProcCtx,
+    source: mgp.Vertex,
+    target: mgp.Vertex,
+    edge_property: mgp.Nullable[str] = None,
+    directed: bool = True,
+) -> mgp.Record(length=float):
+    graph, id_mapping, _ = create_igraph_from_ctx(ctx, directed=directed)
+
+    return mgp.Record(
+        length=graph.shortest_paths(
+            source=id_mapping[source.id],
+            target=id_mapping[target.id],
+            weights=edge_property,
+        )[0][0]
+    )
+
+
+@mgp.read_proc
+def all_shortest_path_lengths(
+    ctx: mgp.ProcCtx,
+    edge_property: mgp.Nullable[str] = None,
+    directed: bool = False,
+) -> mgp.Record(lengths=List[List[float]]):
+    graph, _, _ = create_igraph_from_ctx(ctx, directed=directed)
+
+    return mgp.Record(
+        lengths=graph.shortest_paths(
+            weights=edge_property,
+        )
+    )
+
+
+@mgp.read_proc
+def get_shortest_path(
+    ctx: mgp.ProcCtx,
+    source: mgp.Vertex,
+    target: mgp.Vertex,
+    edge_property: mgp.Nullable[str] = None,
+    directed: bool = True,
+) -> mgp.Record(path=List[mgp.Vertex]):
+    graph, id_mapping, invert_id_mapping = create_igraph_from_ctx(
+        ctx, directed=directed
+    )
+    path = graph.get_shortest_paths(
+        v=id_mapping[source.id], to=id_mapping[target.id], weights=edge_property
+    )[0]
```

Comment:
How is the output of `get_shortest_paths` structured, why do we take only the first element?

Reply:
We take only first element because they use same function for finding shortest path between source and one destination and for finding shortest path between source and multiple destinations. Function returns list that contains shortest path between each source node and destination node.

Reply:
It would be beneficial to return the Path object, not only the ids of path nodes

Reply:
Currently I'm returning list of mgp.Vertex not just vertex ids. I can convert it to mgp.Path with only vertices without edges, if you think there is a difference.